cmake/alpakaCommon.cmake

```diff
@@ -467,173 +459,49 @@ endif()
-                endforeach()
-
-                if(ALPAKA_FAST_MATH STREQUAL ON)
-                    alpaka_set_compiler_options(DEVICE list alpaka "--use_fast_math")
-                endif()
-
-                if(ALPAKA_FTZ STREQUAL ON)
-                    alpaka_set_compiler_options(DEVICE list HIP_HIPCC_FLAGS --ftz=true)
-                elseif(ALPAKA_FTZ STREQUAL OFF)
-                    alpaka_set_compiler_options(DEVICE list HIP_HIPCC_FLAGS --ftz=false)
-                endif()
-
-                if(ALPAKA_HIP_SHOW_REGISTER STREQUAL ON)
-                    alpaka_set_compiler_options(DEVICE list HIP_HIPCC_FLAGS -Xptxas=-v)
-                endif()
-
-                # avoids warnings on host-device signatured, default constructors/destructors
-                alpaka_set_compiler_options(DEVICE list HIP_HIPCC_FLAGS -Xcudafe=--diag_suppress=esa_on_defaulted_function_ignored)
-
-                # random numbers library ( HIP(NVCC) ) /hiprand
-                # HIP_ROOT_DIR is set by FindHIP.cmake
-                find_path(HIP_RAND_INC
-                    NAMES "hiprand_kernel.h"
-                    PATHS "${HIP_ROOT_DIR}/hiprand" "${HIP_ROOT_DIR}/include" "hiprand"
-                    PATHS "/opt/rocm/rocrand/hiprand"
-                    PATH_SUFFIXES "include" "hiprand")
-                find_library(HIP_RAND_LIBRARY
-                    NAMES "hiprand-d" "hiprand"
-                    PATHS "${HIP_ROOT_DIR}/hiprand" "${HIP_ROOT_DIR}" "hiprand"
-                    PATHS "/opt/rocm/rocrand/hiprand"
-                    ENV HIP_PATH
-                    PATH_SUFFIXES "lib" "lib64")
-                if(NOT HIP_RAND_INC)
-                    message(FATAL_ERROR "Could not find hipRAND include (also searched in: HIP_ROOT_DIR=${HIP_ROOT_DIR}).")
-                endif()
-                if(NOT HIP_RAND_LIBRARY)
-                    message(FATAL_ERROR "Could not find hipRAND library (also searched in: HIP_ROOT_DIR=${HIP_ROOT_DIR}).")
-                endif()
-                target_include_directories(alpaka INTERFACE ${HIP_RAND_INC})
-                target_link_libraries(alpaka INTERFACE ${HIP_RAND_LIBRARY})
-            elseif(ALPAKA_HIP_PLATFORM MATCHES "clang")
-                # # hiprand requires ROCm implementation of random numbers by rocrand
-                find_package(rocrand REQUIRED CONFIG
-                    HINTS "${HIP_ROOT_DIR}/rocrand"
-                    HINTS "/opt/rocm/rocrand")
-                if(rocrand_FOUND)
-                    target_include_directories(alpaka INTERFACE ${rocrand_INCLUDE_DIRS})
-                    # ATTENTION: rocRand libraries are not required by alpaka
-                else()
-                    MESSAGE(FATAL_ERROR "Could not find rocRAND (also searched in: HIP_ROOT_DIR=${HIP_ROOT_DIR}/rocrand).")
-                endif()
+        alpaka_compiler_option(HIP_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps 'CMakeFiles/<targetname>.dir'" OFF)
```

Comment:
Could we unify this with the CUDA KEEP FILES? Or not, because CUDA is now using CMake native CUDA support?

Reply:
It can be potentially unified with CUDA but not with all CPU backends. Never the less I would not integrate it into this PR because this is out of the scope.